### PR TITLE
[FEATURE] Add switch theme drop down menu

### LINKED ISSUE #2: [FEATURE] Switch Theme Drop-down menu
### Describe the feature you want 

- A drop down menu for user to choose between different color themes between “black", "dark", "light", "auto".

### Useful reference

The change theme button of the Eureka theme is a good example.

https://www.wangchucheng.com/en/docs/hugo-eureka/

### PR DESCRIPTION
**Pull request by HEIGE-PCloud**
- Add a button for a select list to select different color theme
- Add a option to change between select and switch in config.toml

### CODE REVIEW

File: layouts/partials/header.html [hunk 1]
```diff
@@ -79,14 +79,46 @@
...
+                    <select class="theme-select" id="theme-select-desktop" onchange="setTheme(this.value);" title="{{ T "switchTheme" }}">
+                        <option value="light">Light</option>
+                        <option value="dark">Dark</option>
+                        <option value="black">Black</option>
+                        <option value="auto">Auto</option>
                     </select>
-                </a> -->
+                </a>
+                <script>
+                    function getTheme() {
```
Comment (HEIGE-PCloud) [inline layouts/partials/header.html]: The script needs to put in theme.js and update `this.isDark` to update the theme of some shortcodes.

File: layouts/partials/header.html [hunk 2]
```diff
@@ -76,17 +76,21 @@
...
                     </span>
                 {{- end -}}
+                {{- if eq .Site.Params.header.themeChangeMode "select" -}}
+                <a href="javascript:void(0);" class="menu-item theme-select" title="{{ T "switchTheme" }}">
+                    <i class="fas fa-adjust fa-fw"></i>
+                    <select class="color-theme-select" id="theme-select-desktop" title="{{ T "switchTheme" }}">
+                        <option value="light">Light</option>
+                        <option value="dark">Dark</option>
+                        <option value="black">Black</option>
+                        <option value="auto">Auto</option>
```
Comment (HEIGE-PCloud) [inline layouts/partials/header.html]: Need local translation for `value`.

File: layouts/partials/header.html [hunk 3]
```diff
@@ -154,9 +158,21 @@
...
                 </a>
             {{- end -}}
+            {{- if eq .Site.Params.header.themeChangeMode "select" -}}
+            <a href="javascript:void(0);" class="menu-item theme-select" title="{{ T "switchTheme" }}">
+                <i class="fas fa-adjust fa-fw"></i>
+                <select class="color-theme-select" id="theme-select-mobile" title="{{ T "switchTheme" }}">
+                    <option value="light">Light</option>
+                    <option value="dark">Dark</option>
+                    <option value="black">Black</option>
+                    <option value="auto">Auto</option>
```
Comment (HEIGE-PCloud) [inline layouts/partials/header.html]: Same issue here, need translation.

**Review by SpongeBab — APPROVED** [review]

**Review by SpongeBab — APPROVED** [review]

**Comment by HEIGE-PCloud** [comment]
Notice that outdated cache may cause issue when enable the `themeChangeMode` for the first time.

**Pull request merged**